templates/flutter/lib/src/realtime_mixin.dart.twig

```diff
@@ -15,35 +15,46 @@ typedef GetFallbackCookie = String? Function();
 
 mixin RealtimeMixin {
   late Client client;
-  final Map<String, List<StreamController<RealtimeMessage>>> _channels = {};
+  final Set<String> _channels = {};
   WebSocketChannel? _websok;
   String? _lastUrl;
   late WebSocketFactory getWebSocket;
   GetFallbackCookie? getFallbackCookie;
   int? get closeCode => _websok?.closeCode;
+  int _subscriptionsCounter = 0;
+  Map<int, RealtimeSubscription> _subscriptions = {};
```

Comment:
Now I'm getting a concurrent modification error:

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Concurrent modification during iteration: _Map len:4.
#0      _CompactIterator.moveNext (dart:collection-patch/compact_hash.dart:714:7)
#1      RealtimeMixin._createSocket.<anonymous closure> (package:appwrite/src/realtime_mixin.dart:92:49)
#2      _RootZone.runGuarded (dart:async/zone.dart:1582:10)
#3      _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:392:13)
#4      _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:402:7)
#5      _BufferingStreamSubscription._close (dart:async/stream_impl.dart:291:7)
#6      _ForwardingStream._handleDone (dart:async/stream_pipe.dart:99:10)
#7      _ForwardingStreamSubscription._handleDone (dart:async/stream_pipe.dart:161:13)
#8      _RootZone.runGuarded (dart:async/zone.dart:1582:10)
#9      _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:392:13)
#10     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.d<…>
```

Here's the code I tested with: https://github.com/appwrite/playground-for-flutter/pull/46. The list documents returns 4 documents and then i subscribe to each of them.

Maybe we ignore this though because I'm iterating over the documents and subscribing to all of them?

Reply:
Hmmm even subscribing to 1 channel triggers the error 😭 

<img width="1170" alt="image" src="https://github.com/appwrite/sdk-generator/assets/1477010/7c71883f-5ad0-41d4-9dd0-7e6d0e0e9754">

Reply:
Actually... this might be happening due to unreliable realtime connection causing the realtime connection to disconnect and then this being executed:

https://github.com/appwrite/sdk-generator/blob/18ad858ea12af4717263b59a8bc18b663f03ad75/templates/flutter/lib/src/realtime_mixin.dart.twig#L92-L94

Although, this part might be problematic in general...

Reply:
btw, this is on an iOS simulator

Reply:
Why problematic, we need to close the subscriptions once done, otherwise the subscribers of the streams will never no when it was closed or disconnected in some way

Reply:
My be my latest push will help, before closing the subscription on done, I also check if something else is creating socket again i.e. `_creatingSocket` is true. That should prevent concurrent modification. Let me know if that works.